### DISCUSSION TITLE
Calculating (Cs)neg and (Cs)pos at each SOC point

### DISCUSSION THREAD

**Discussion by kumaryash7** [pybamm-team/PyBaMM#4008] [Q&A]
I see there is an option to plot the concentration of positive and negative electrodes during charging/discharging versus capacity/SOC.
I want to know the **calculation/maths to calculate the Cs neg and Cs pos values at any SOC**. I know both electrodes' (Cs)max for pos, neg, and initial conc.
Suppose I want to know the (Cs)pos and (Cs)neg values at 0.3 SOC.?
TIA.

**Comment by abhishek-appana** [comment]
I doubt there is a straightforward way to calculate the lithium concentration at a particular SOC. You need to use the whole electrochemical model, solving it at each time step to get these values!

The math for these calculations depends on the electrochemical model you choose. See the types of models in the link (https://docs.pybamm.org/en/stable/source/api/models/lithium_ion/index.html)

**Comment by rtimms** [comment]
At equilibrium, see [this example](https://github.com/pybamm-team/PyBaMM/blob/develop/docs/source/examples/notebooks/models/electrode-state-of-health.ipynb).
